packages/ui/components/input-file/src/LionInputFile.js

```diff
@@ -481,15 +498,17 @@ export class LionInputFile extends ScopedElementsMixin(LocalizeMixin(LionField))
     }
 
     this._inputNode.files = ev.dataTransfer.files;
+
+    if (this.multiple) {
+      const computedFiles = this.__computeNewAddedFiles(
+        /** @type {InputFile[]} */ (Array.from(ev.dataTransfer.files)),
+      );
+      this.modelValue = [...(this.modelValue ?? []), ...computedFiles];
+    } else {
+      this.modelValue = Array.from(ev.dataTransfer.files);
+    }
+
     // @ts-ignore
```

Comment:
I guess this line can also be removed

Reply:
This one was required due to typing, but now it's replaced with type casting comment.